shared/macros-bash.jinja

```diff
@@ -686,3 +686,75 @@ dpkg-query --show --showformat='${db:Status-Status}\n' "{{{ pkgname }}}" 2>/dev/
 rpm --quiet -q "{{{ pkgname }}}"
 {{%- endif -%}}
 {{%- endmacro -%}}
+
+{{#
+  Macro to replace configuration setting in config file or add the configuration setting if
+  it does not exist.
+ 
+  Expects arguments:
+ 
+  config_file:		Configuration file that will be modified
+  key:			Configuration option to change
+  value:		Value of the configuration option to change
+  cce:			The CCE identifier or '@CCENUM@' if no CCE identifier exists
+  format:		The printf-like format string that will be given stripped key and value as arguments,
+ 			so e.g. '%s=%s' will result in key=value subsitution (i.e. without spaces around =)
+ 
+  Optional arugments:
+ 
+  format:		Optional argument to specify the format of how key/value should be
+  			modified/appended in the configuration file. The default is key = value.
+ 
+  Example Call(s):
+ 
+      With default format of 'key = value':
+      {{{ bash_replace_or_append('/etc/sysctl.conf', '^kernel.randomize_va_space', '2', '@CCENUM@') }}}
+ 
+      With custom key/value format:
+      {{{ bash_replace_or_append('/etc/sysconfig/selinux', '^SELINUX=', 'disabled', '@CCENUM@', '%s=%s') }}}
+ 
+      With a variable:
+      {{{ bash_replace_or_append('/etc/sysconfig/selinux', '^SELINUX=', "$var_selinux_state", '@CCENUM@', '%s=%s') }}}
+ 
+#}}
+{{%- macro bash_replace_or_append(config_file, key, value, cce, format='%s = %s') -%}}
+case_insensitive_mode=yes sed_case_insensitive_option='' grep_case_insensitive_option=''
+
+if [ "$case_insensitive_mode" = yes ]; then
+    sed_case_insensitive_option="i"
+    grep_case_insensitive_option="-i"
+fi
```

Comment:
Just out of curiosity, what is this? It looks like that we are always in case insensitive mode, and therefore `sed_case_insensitive_option` and `grep_case_insensitive_option` could be macros instead of shell variables.

Reply:
Nice finding! It was there before. Case insensitive mode is always yes.  So I guess it could be removed completely, and substitute the bash variables by their values.

Reply:
Answering to myself - #2689  explains that. Therefore, I suggest renaming the macro to `replace_or_append_case_insensitive`